apps/client/src/pages/QuestionsPage.js

```diff
@@ -1,40 +1,44 @@
-import React, { useState } from "react";
+import React, { useState, useContext, useEffect } from "react";
 import { useHistory, useParams, Redirect } from "react-router-dom";
-import { geturl, routes, getslug } from "../routes";
 import { Helmet } from "react-helmet";
 
 import withChecker from "../hoc/withChecker";
+import { SessionContext } from "../context";
+import { geturl, routes, getslug } from "../routes";
 import Layout from "../components/Layouts/DefaultLayout";
 import DebugDecisionTable from "../components/DebugDecisionTable";
 import Question, { booleanOptions } from "../components/Question";
-// import ErrorPage from "./ErrorPage";
 
 const QuestionsPage = ({ topic, checker }) => {
+  const sessionContext = useContext(SessionContext);
   const params = useParams();
   const history = useHistory();
+  const { question: questionSlug } = params;
+
   const [question, setQuestion] = useState(
     checker.stack[checker.stack.length - 1]
   );
+  const { slug } = topic;
 
-  const { question: questionSlug } = params;
-  const currSlug = getslug(question.text);
+  useEffect(() => {
+    if (!questionSlug) {
+      sessionContext.setSessionData({
+        answers: checker.getQuestionAnswers(),
+      });
+    }
+  });
 
   // Update URL based on question text
-  if (!questionSlug || questionSlug !== currSlug) {
+  if (!questionSlug) {
```

Comment:
why remove the logic to redirect if the url-slug does not correspond with the question-slug?

Reply:
I have no idea, I did not touch this. I will put it back for now. 

@svenjens do you know?

Reply:
Fixed